### PR TITLE
Soumyadeep/conf sauvola

### PR DESCRIPTION
**Pull request by soumyadeep9**
### Description

We propose a novel binarization technique that improves upon the Sauvola's algorithm by preserving more foreground information in the binarized document-images. The computational complexity of the proposed algorithm is the same as the Sauvola's algorithm, thereby making it suitable even for mobile-computing based applications.  

### Checklist:

- [ x] I have followed the [contribution guidelines](../CONTRIBUTING.md) and code style for this project.
- [x ] I have added tests covering my contributions.
- [x ] I have updated the documentation accordingly.
- [ x] This PR is being made to `staging` and not `master`
- [ x] I will squash merge this PR into `staging`

### CODE REVIEW

File: contrib/binarization/confidence_based_Sauvola_binarization/README.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+### Confidence based Sauvola outputs
+
+<img src="./results/2_bin_new.png" width="33%"> </img>
+<img src="./results/10_bin_new.png" width="33%"> </img>
+<img src="./results/new1_bin_new.png" width="33%"> </img>
+
+## Reference
+
+For details refer to this [paper](./ModifiedSauvola.pdf)
```
Comment (PatrickBue) [inline contrib/binarization/confidence_based_Sauvola_binarization/README.md:27]: Should this simply point to the paper in this folder called "ModifiedSauvola.pdf" instead?

**Review by PatrickBue — COMMENTED** [review]
Looks great! Left one comment in the code, and one more comment here:
- The directory __pycache__ is likely not checked in on purpose and can get deleted.

File: contrib/binarization/confidence_based_Sauvola_binarization/README.md (hunk 1 above)
Comment (soumyadeep9) [reply contrib/binarization/confidence_based_Sauvola_binarization/README.md:27]: Here we use ./ModifiedSauvola.pdf to pick the pdf file from the current working directory only.

**Comment by PatrickBue** [comment]
@soumyadeep9 Thanks for updating the pdf link. Please also remove the \_\_pycache\_\_ directory since I guess it's not needed, and I will merge then. Looks great!!

**Comment by soumyadeep9** [comment]
> @soumyadeep9 Thanks for updating the pdf link. Please also remove the __pycache__ directory since I guess it's not needed, and I will merge then. Looks great!!

It is done

**Review by PatrickBue — APPROVED** [review]

**Comment by PatrickBue** [comment]
> > @soumyadeep9 Thanks for updating the pdf link. Please also remove the **pycache** directory since I guess it's not needed, and I will merge then. Looks great!!
> 
> It is done

Great! It's merged now into staging! Great contribution!

**Pull request merged**